### PR TITLE
bug 1483670: accept unicode from update endpoints

### PR DESCRIPTION
**Pull request by mozbhearsum**
This is a follow-up to #634 and #744 that allows us to accept unicode in all of the string fields for the /update endpoints. Because we receive so many weird requests, I don't think we can get away with throwing errors for them all. This patch accepts it - but replaces the unicode characters with "?". In a far flung future where we support unicode at the db and other balrog layers, we should be able to simply strip this out, or perhaps encode to utf-8 instead. In addition to the unit tests included, I also grabbed a bunch of URLs from Sentry and verified that I got 200s from them after applying this patch.

Finally, I copied over the UnicodeEncodeError handler from the admin app to public. Although the /update endpoints don't support unicode, a couple of fields on the public API do - so that handler is still useful for them.

### CODE REVIEW

**Comment by nthomas-mozilla** [comment]
This looks good to me. A closely related issue is tracebacks like at https://sentry.prod.mozaws.net/operations/prod-public/issues/4339330/, where there is a bogus unicode query parameter. Would it be worth addressing this at the same time ?

**Comment by mozbhearsum** [comment]
Definitely. I'll update this PR to handle them.

**Comment by mozbhearsum** [comment]
OK, I've updated this to handle the query args as best as possible (accept unicode as values, and verify that unicode in field names doesn't throw a 500).

I also updated this to fix some issues found by the new Python 3 tests (thanks @allan-silva!). I think there may be more work to do before we upgrade to Python 3, but this should be OK for now.

File: version.txt [hunk 1]
```diff
@@ -1 +1 @@
-2.53
\ No newline at end of file
+2.53
```
[the remark this answers is not in the document]
Comment (nthomas-mozilla) [reply version.txt]: Ah, you're fixing up my removal of the newline, never mind!

**Pull request merged**